repl/history.go

```diff
@@ -0,0 +1,111 @@
+package repl
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/abs-lang/abs/util"
+)
+
+/*
+Support for abs history file in the interactive REPL:
+
+1) The current ABS_HISTORY_FILE is loaded into the prompt.Run() cycle
+   using prompt.OptionHistory(getHistory()). This also loads the local history as well.
+   Default ABS_HISTORY_FILE is "~/.abs_history".
+2) Append each non-null, unique next line passed from prompt to the executor() to the local history.
+   NB. the live prompt history shows duplicate next lines, but they are not saved to the local history.
+3) Save the local history whenever the prompt.Run() loop exits (^D) or the executor() exits (on quit).
+   Write the local history to the ABS_HISTORY_FILE up to ABS_MAX_HISTORY_LINES (default 1000 lines).
+
+Note that ABS_HISTORY_FILE and ABS_MAX_HISTORY_LINES variables may come from the OS environment.
+*/
+
+const (
+	ABS_HISTORY_FILE      = "~/.abs_history"
+	ABS_MAX_HISTORY_LINES = "1000"
+)
+
+// expand full path to ABS_HISTORY_FILE for current user and get ABS_MAX_HISTORY_LINES
+func getHistoryConfiguration() (string, int) {
+	// obtain any OS environment variables
+	// ABS_MAX_HISTORY_LINES
+	maxHistoryLines := os.Getenv("ABS_MAX_HISTORY_LINES")
+	if len(maxHistoryLines) == 0 {
+		maxHistoryLines = ABS_MAX_HISTORY_LINES
+	}
+	maxLines, _ := strconv.Atoi(maxHistoryLines)
+	// ABS_HISTORY_FILE
+	historyFile := os.Getenv("ABS_HISTORY_FILE")
+	if len(historyFile) == 0 {
+		historyFile = ABS_HISTORY_FILE
+	}
+	if maxLines > 0 {
+		// expand the ABS_HISTORY_FILE to the user's HomeDir
+		filePath, err := util.ExpandPath(historyFile)
+		if err != nil {
+			os.Exit(99)
+		}
+		historyFile = filePath
+	} else {
+		historyFile = ""
```

Comment:
In what case would we end here? Looks funky :)

Reply:
This case is overkill. When `maxLines == 0`, all file operations are suppressed, so the filename is irrelevant.

I will remove it.